### PR TITLE
Increase test code coverage

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

**Review by hiqsol — APPROVED** [review]

File: tests/FileRotatorTest.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        ]);
+
+        $logger->debug(str_repeat('x', 1024));
+        $logger->flush(true);
+        self::assertFileExists($logFile);
+        self::assertFileDoesNotExist($logFile . '.1');
+
+        $nonRotatedFileContent = file_get_contents($logFile);
+
+        $logger->debug("\0");
```
Comment (samdark) [inline tests/FileRotatorTest.php:36]: Why null-byte instead of another `x`? Any special meaning here?

File: tests/FileRotatorTest.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        $logFile = $this->getLogFilePath();
+        $rotator = new FileRotator(1, 2, 777, $rotateByCopy);
+        $fileTarget = new FileTarget($logFile, $rotator);
+
+        $logger = new Logger([
+            'file' => $fileTarget,
+        ]);
+
+        $logger->debug(str_repeat('x', 1024));
```
Comment (samdark) [inline tests/FileRotatorTest.php:29]:
```suggestion
        $oneKilobyteOfData = str_repeat('x', 1024);
        $logger->debug($oneKilobyteOfData);
```

File: tests/FileRotatorTest.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        $rotator = new FileRotator(1, $filesCount, null, true);
+        $fileTarget = new FileTarget($logFile, $rotator);
+
+        $logger = new Logger([
+            'file' => $fileTarget,
+        ]);
+
+        foreach (range(1, $filesCount+2) as $i) {
+            /** @noinspection DisconnectedForeachInstructionInspection */
+            $logger->debug(str_repeat('x', 1024));
```
Comment (samdark) [inline tests/FileRotatorTest.php:61]:
```suggestion
            $oneKilobyteOfData = str_repeat('x', 1024);
            $logger->debug($oneKilobyteOfData);
```

File: tests/FileRotatorTest.php [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        $logFile = $this->getLogFilePath();
+        $rotator = new FileRotator(1, $filesCount, null, true);
+        $fileTarget = new FileTarget($logFile, $rotator);
+
+        $logger = new Logger([
+            'file' => $fileTarget,
+        ]);
+
+        foreach (range(1, $filesCount+2) as $i) {
```
Comment (samdark) [inline tests/FileRotatorTest.php:59]: `while` may be a bit easier to read in this case...

File: tests/FileRotatorTest.php [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        $logger->flush(true);
+
+        self::assertFileExists($logFile);
+        self::assertFileExists($logFile . '.1');
+
+        self::assertEquals($nonRotatedFileContent, file_get_contents($logFile . '.1'));
+    }
+
+    /**
+     * @dataProvider filesCountProvider()
```
Comment (samdark) [inline tests/FileRotatorTest.php:46]: Do we need different number of files? Does coverage change if we always use 3?

File: tests/FileRotatorTest.php [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        foreach (range(1, $filesCount+2) as $i) {
+            /** @noinspection DisconnectedForeachInstructionInspection */
+            $logger->debug(str_repeat('x', 1024));
+            /** @noinspection DisconnectedForeachInstructionInspection */
+            $logger->flush(true);
+        }
+
+        self::assertFileExists($logFile);
+
+        foreach (range(1, $filesCount) as $counter) {
```
Comment (samdark) [inline tests/FileRotatorTest.php:68]: `while` may be a bit easier to read in this case...

File: tests/FileRotatorTest.php [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+            $filesName = $counter !== 1 ? $logFile . '.' . ($counter - 1) : $logFile;
+            self::assertFileExists($filesName);
+        }
+
+        // check that next file is not exist
```
Comment (samdark) [inline tests/FileRotatorTest.php:73]:
```suggestion
        // check that next file does not exist
```

File: tests/FileRotatorTest.php [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        $filesCount++;
+        self::assertFileDoesNotExist("{$logFile}.{$filesCount}");
+    }
+
+    /**
+     * @dataProvider filesSizesProvider()
```
Comment (samdark) [inline tests/FileRotatorTest.php:79]: Do we need all these sizes? Maybe one size is enough?

File: tests/FileRotatorTest.php [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        self::assertGreaterThan($maxFileSize, filesize($logFile) / 1024);
+    }
+
+    public function testDefaultMaxFileSize(): void
+    {
+        $rotator = new FileRotator();
+        self::assertEquals(10240, $rotator->getMaxFileSize());
+    }
+
+    public function testMaxFileSizeLowerThanOne(): void
```
Comment (samdark) [inline tests/FileRotatorTest.php:106]: How about throwing an exception in this case instead?

File: tests/FileRotatorTest.php [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        $rotator = new FileRotator(-1);
+        self::assertEquals(1, $rotator->getMaxFileSize());
+    }
+
+    public function testSetMaxFileSizeLowerThanOne(): void
```
Comment (samdark) [inline tests/FileRotatorTest.php:112]: How about throwing an exception in this case instead?

File: tests/FileRotatorTest.php [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+        self::assertEquals(1, $rotator->getMaxFileSize());
+    }
+
+    public function testDefaultMaxFiles(): void
+    {
+        $rotator = new FileRotator();
+        self::assertEquals(5, $rotator->getMaxFiles());
+    }
+
+    public function testMaxFilesLowerThanOne(): void
```
Comment (samdark) [inline tests/FileRotatorTest.php:125]: How about throwing an exception in this case instead?

File: tests/FileRotatorTest.php [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        $rotator = new FileRotator(0, -1);
+        self::assertEquals(1, $rotator->getMaxFiles());
+    }
+
+    public function testSetMaxFilesLowerThanOne(): void
```
Comment (samdark) [inline tests/FileRotatorTest.php:131]: How about throwing an exception in this case instead?

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by roxblnfk — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**